Libs/vtkAddon/vtkPersonInformation.cxx

```diff
@@ -0,0 +1,175 @@
+vtkGetSetStringInformationCxxMacro(Login);
+vtkGetSetStringInformationCxxMacro(Organization);
+vtkGetSetStringInformationCxxMacro(OrganizationRole);
+vtkGetSetStringInformationCxxMacro(ProcedureRole);
+
+//----------------------------------------------------------------------------
+vtkPersonInformation::vtkPersonInformation()
+{
+}
+
+//----------------------------------------------------------------------------
+vtkPersonInformation::~vtkPersonInformation()
+{
+}
+
+//----------------------------------------------------------------------------
+void vtkPersonInformation::PrintSelf(ostream& os, vtkIndent indent)
+{
+  this->Superclass::PrintSelf(os,indent);
+
+  for (std::map<std::string, std::string>::iterator it = this->Data.begin();
+    it != this->Data.end(); ++it)
+    {
+    os << indent << it->first << ": " << it->second << "\n";
+  }
+}
+
+//----------------------------------------------------------------------------
+void vtkPersonInformation::DeepCopy(vtkPersonInformation* source)
+{
+  if (!source)
+    {
+    return;
+    }
+  this->Data = source->Data;
+}
+
+//----------------------------------------------------------------------------
+void vtkPersonInformation::SetCustomString(const std::string& key, const std::string& value)
+{
+  std::string currentValue = this->Data[key];
+  if (value.compare(currentValue) == 0)
+    {
+    // no change
+    return;
+    }
+  this->Data[key] = value;
+  this->Modified();
+}
+
+//----------------------------------------------------------------------------
+std::string vtkPersonInformation::GetCustomString(const std::string& key)
+{
+  std::map<std::string, std::string>::iterator it = this->Data.find(key);
+  if (it == this->Data.end())
+    {
+    return "";
+    }
+  return it->second;
+}
+
+//----------------------------------------------------------------------------
+std::string vtkPersonInformation::WriteToString()
+{
+  std::string output;
+  for (std::map<std::string, std::string>::iterator it = this->Data.begin();
+    it != this->Data.end(); ++it)
+    {
+    if (!output.empty())
+      {
+      output += ";";
+      }
+    output += this->EncodeString(it->first) + ":" + this->EncodeString(it->second);
+    }
+  return output;
+}
+
+//----------------------------------------------------------------------------
+bool vtkPersonInformation::ReadFromString(const std::string& data)
+{
+  this->Data.clear();
+  std::stringstream keyValuePairs(data);
+  std::string keyValuePair;
+  while (std::getline(keyValuePairs, keyValuePair, ';'))
+    {
+    int colonIndex = keyValuePair.find(':');
+    std::string key = this->DecodeString(keyValuePair.substr(0, colonIndex));
+    std::string value = this->DecodeString(keyValuePair.substr(colonIndex + 1));
+    this->SetCustomString(key, value);
+    }
+  return true;
+}
+
+//----------------------------------------------------------------------------
+std::string vtkPersonInformation::EncodeString(const std::string& value)
```

Comment:
We have a generic url encode/decode function available vtkMRMLNode.  I wonder if that should move to vtkAddon and then be used from here and from mrml.

https://github.com/Slicer/Slicer/blob/8198bffc09899bc55140f6f375452ace4714c598/Libs/MRML/Core/vtkMRMLNode.cxx#L931

Reply:
There are already two encoders (URL and XML) in vtkMRMLNode, yet none of them are suitable for encoding these kind of property lists (where special characters are ```;``` and ```:```). The XML encoder would be usable if we stored the person information in an XML tree, which would be very flexible, but seems that it would be somewhat an overkill.

Regardless, we could move XML, URL, and generic property list encoders/decoders to vtkAddon.

Reply:
Ah, so this is specific to property lists.  I guess that could be more explicit in the method name but otherwise okay to keep as a method here.